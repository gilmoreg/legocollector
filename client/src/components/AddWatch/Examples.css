.Examples h4 {
    font-size: 14px;
    margin-bottom: 0.25rem;
}

.Examples ul {
    font-size: 12px;
    margin-top: 0.25rem;
    margin-left: 0;
    text-align: left;
    list-style: none;
}

/* .Examples button {
    text-align: left;
}

.Examples button:hover {
    color: #FF7F66;
} */