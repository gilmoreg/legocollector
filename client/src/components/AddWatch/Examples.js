import React from 'react';
import PropTypes from 'prop-types';
import './Examples.css';

const Examples = (props) => (
  <div className="Examples">
    <h4>Examples (click to test out):</h4>
    <ul>
      <li key="75105">
        <button data-id={75105} onClick={props.click}>75105 - Star Wars Millenium Falcon</button>
      </li>
      <li key="21029">
        <button data-id={21029} onClick={props.click}>21029 - Architecture Buckingham Palace</button>
      </li>
      <li key="10232">
        <button data-id={10232} onClick={props.click}>10232 - Modular Buildings Palace Cinema</button>
      </li>
      <li key="10217">
        <button data-id={10217} onClick={props.click}>10217 - Harry Potter Diagon Alley</button>
      </li>
    </ul>
  </div>
);

Examples.propTypes = {
  click: PropTypes.func.isRequired,
}

export default Examples;
