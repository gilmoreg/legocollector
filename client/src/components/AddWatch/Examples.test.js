import React from 'react';
import { shallow } from 'enzyme';
import Examples from './Examples';

it('renders without crashing', () => {
  const wrapper = shallow(
    <Examples click={() => {}} />);
  expect(wrapper).toMatchSnapshot();
});
