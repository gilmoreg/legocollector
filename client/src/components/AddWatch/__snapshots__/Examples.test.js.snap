// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Examples
    click={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <styled.h4>
          Examples (click to test out):
        </styled.h4>,
        <styled.ul>
          <li>
            <styled.button
              data-id={75105}
              onClick={[Function]}
            >
              75105 - Star Wars Millenium Falcon
            </styled.button>
          </li>
          <li>
            <styled.button
              data-id={21029}
              onClick={[Function]}
            >
              21029 - Architecture Buckingham Palace
            </styled.button>
          </li>
          <li>
            <styled.button
              data-id={10232}
              onClick={[Function]}
            >
              10232 - Modular Buildings Palace Cinema
            </styled.button>
          </li>
          <li>
            <styled.button
              data-id={10217}
              onClick={[Function]}
            >
              10217 - Harry Potter Diagon Alley
            </styled.button>
          </li>
        </styled.ul>,
      ],
      "className": "Examples",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": "Examples (click to test out):",
        },
        "ref": null,
        "rendered": "Examples (click to test out):",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <li>
              <styled.button
                data-id={75105}
                onClick={[Function]}
              >
                75105 - Star Wars Millenium Falcon
              </styled.button>
            </li>,
            <li>
              <styled.button
                data-id={21029}
                onClick={[Function]}
              >
                21029 - Architecture Buckingham Palace
              </styled.button>
            </li>,
            <li>
              <styled.button
                data-id={10232}
                onClick={[Function]}
              >
                10232 - Modular Buildings Palace Cinema
              </styled.button>
            </li>,
            <li>
              <styled.button
                data-id={10217}
                onClick={[Function]}
              >
                10217 - Harry Potter Diagon Alley
              </styled.button>
            </li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "75105",
            "nodeType": "host",
            "props": Object {
              "children": <styled.button
                data-id={75105}
                onClick={[Function]}
              >
                75105 - Star Wars Millenium Falcon
              </styled.button>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": "75105 - Star Wars Millenium Falcon",
                "data-id": 75105,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "75105 - Star Wars Millenium Falcon",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "21029",
            "nodeType": "host",
            "props": Object {
              "children": <styled.button
                data-id={21029}
                onClick={[Function]}
              >
                21029 - Architecture Buckingham Palace
              </styled.button>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": "21029 - Architecture Buckingham Palace",
                "data-id": 21029,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "21029 - Architecture Buckingham Palace",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "10232",
            "nodeType": "host",
            "props": Object {
              "children": <styled.button
                data-id={10232}
                onClick={[Function]}
              >
                10232 - Modular Buildings Palace Cinema
              </styled.button>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": "10232 - Modular Buildings Palace Cinema",
                "data-id": 10232,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "10232 - Modular Buildings Palace Cinema",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "10217",
            "nodeType": "host",
            "props": Object {
              "children": <styled.button
                data-id={10217}
                onClick={[Function]}
              >
                10217 - Harry Potter Diagon Alley
              </styled.button>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": "10217 - Harry Potter Diagon Alley",
                "data-id": 10217,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "10217 - Harry Potter Diagon Alley",
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <styled.h4>
            Examples (click to test out):
          </styled.h4>,
          <styled.ul>
            <li>
              <styled.button
                data-id={75105}
                onClick={[Function]}
              >
                75105 - Star Wars Millenium Falcon
              </styled.button>
            </li>
            <li>
              <styled.button
                data-id={21029}
                onClick={[Function]}
              >
                21029 - Architecture Buckingham Palace
              </styled.button>
            </li>
            <li>
              <styled.button
                data-id={10232}
                onClick={[Function]}
              >
                10232 - Modular Buildings Palace Cinema
              </styled.button>
            </li>
            <li>
              <styled.button
                data-id={10217}
                onClick={[Function]}
              >
                10217 - Harry Potter Diagon Alley
              </styled.button>
            </li>
          </styled.ul>,
        ],
        "className": "Examples",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": "Examples (click to test out):",
          },
          "ref": null,
          "rendered": "Examples (click to test out):",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <li>
                <styled.button
                  data-id={75105}
                  onClick={[Function]}
                >
                  75105 - Star Wars Millenium Falcon
                </styled.button>
              </li>,
              <li>
                <styled.button
                  data-id={21029}
                  onClick={[Function]}
                >
                  21029 - Architecture Buckingham Palace
                </styled.button>
              </li>,
              <li>
                <styled.button
                  data-id={10232}
                  onClick={[Function]}
                >
                  10232 - Modular Buildings Palace Cinema
                </styled.button>
              </li>,
              <li>
                <styled.button
                  data-id={10217}
                  onClick={[Function]}
                >
                  10217 - Harry Potter Diagon Alley
                </styled.button>
              </li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "75105",
              "nodeType": "host",
              "props": Object {
                "children": <styled.button
                  data-id={75105}
                  onClick={[Function]}
                >
                  75105 - Star Wars Millenium Falcon
                </styled.button>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": "75105 - Star Wars Millenium Falcon",
                  "data-id": 75105,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "75105 - Star Wars Millenium Falcon",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "21029",
              "nodeType": "host",
              "props": Object {
                "children": <styled.button
                  data-id={21029}
                  onClick={[Function]}
                >
                  21029 - Architecture Buckingham Palace
                </styled.button>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": "21029 - Architecture Buckingham Palace",
                  "data-id": 21029,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "21029 - Architecture Buckingham Palace",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "10232",
              "nodeType": "host",
              "props": Object {
                "children": <styled.button
                  data-id={10232}
                  onClick={[Function]}
                >
                  10232 - Modular Buildings Palace Cinema
                </styled.button>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": "10232 - Modular Buildings Palace Cinema",
                  "data-id": 10232,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "10232 - Modular Buildings Palace Cinema",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "10217",
              "nodeType": "host",
              "props": Object {
                "children": <styled.button
                  data-id={10217}
                  onClick={[Function]}
                >
                  10217 - Harry Potter Diagon Alley
                </styled.button>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": "10217 - Harry Potter Diagon Alley",
                  "data-id": 10217,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "10217 - Harry Potter Diagon Alley",
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
